### ISSUE TITLE
[plugin|tiling] Monitor Numbering

### ISSUE THREAD

**Issue by koekeishiya** [koekeishiya/chunkwm#262]
Change how chunkwm treats monitors or add some way to tell chunkwm how I expect my monitors to be numbered. Ideally, we would implement something that fixes this automatically, as this is not something that the user should ever really have to deal with.

The most intuitive behaviour (imo), is that the left-most monitor is the first one, and moves increasingly towards the right.

macOS assigns the arrangement indexes as follows, which is currently inherited by chunkwm:

```
+- left monitor -+ +-primary monitor-+ +-right monitor-+
        2              1 (menu bar)            3
```

Most users would probably expect the following:
```
+- left monitor -+ +-primary monitor-+ +-right monitor-+
        1              2 (menu bar)            3
```

This becomes more complex to figure out in situations where the arrangement consists of monitors that are stacked differently, or intermixed.

**Comment by chrishoage** [comment]
This is 100% my personal opinion, but I think (1) being the monitor with the menu bar makes the most sense. 

(1) is your primary monitor. 


Maybe a query param that would let list the monitors (with resolutions + any other ID information we had) so a user could easily and correctly set up their key bindings. 


They could even set up their key bindings to mean `1` on the keyboard is monitor `2`

I don't think we should override macOS concepts whenever we can help it.

**Comment by koekeishiya** [comment]
I agree with pretty much everything you said, but the `chunkc tiling::monitor -f next | prev` cycles "incorrectly" (not as the user would expect) even though they could fix the *id specific* binds manually.

I suppose the more appropriate way is to extend the query support, and the user can then implement his own cycling.

**Comment by chrishoage** [comment]
Maybe a setting that could register the desired order? I have no concept of the level of effort or complexity around that. 

You bring up a very good point about `prev` `next`

`chunkc set monitor_cycle_order 3 1 2` 

This would override the order in which monitors would cycle with `next` and `prev`

**Comment by dominiklohmann** [comment]
I agree with @chrishoage on this. Remapping indices seems like a good solution. What's important is that every single instance of monitor access needs to go through this index mapping.

If the monitor count is not equal to the amount of indices given through `monitor_cycle_order`, the command should be ignored. That way, I can set `3 1 2` for my 3 monitor setup and still have the same config work when I use my laptop without external monitors. 

Ideally, a user could define multiple orders for multiple monitor counts:
```
# The first one is redundant 
chunkc set monitor_cycle_order 1 
chunkc set monitor_cycle_order 2 1
chunkc set monitor_cycle_order 2 1 3
chunkc set monitor_cycle_order 4 2 1 3
```
